### PR TITLE
Replace deprecated `..index.is_all_dates` in plottery

### PR DESCRIPTION
**Pull request by oysteoh**
This PR adjusting a PR merged too quickly ( #3231 ). The test is
now removed as it is just testing pandas functionality. The util
method is not needed and the call to `..data.index.inferred_type`
is done directly where it should be used.

### CODE REVIEW

File: ert_gui/plottery/plots/ensemble.py [hunk 1]
```diff
@@ -25,7 +24,7 @@ def plot(self, figure, plot_context, case_to_data_map, observation_data):
             data = data.T
 
             if not data.empty:
-                if not index_is_datetime(data):
+                if not data.index.inferred_type == "datetime64":
```
Comment (BjarneHerland) [inline ert_gui/plottery/plots/ensemble.py:27]: `!=`  perhaps?

File: ert_gui/plottery/plots/statistics.py [hunk 2]
```diff
@@ -27,7 +26,7 @@ def plot(self, figure, plot_context, case_to_data_map, _observation_data):
         for case, data in case_to_data_map.items():
             data = data.T
             if not data.empty:
-                if not index_is_datetime(data):
+                if not data.index.inferred_type == "datetime64":
```
Comment (BjarneHerland) [inline ert_gui/plottery/plots/statistics.py:29]: like above

**Review by BjarneHerland — APPROVED** [review]
A few nits but all-in-all good stuff  😄

File: ert_gui/plottery/plots/ensemble.py (hunk 1 above)
Comment (oysteoh) [reply ert_gui/plottery/plots/ensemble.py]: YES! absolutely... :see_no_evil:

**Pull request merged**